android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -645,6 +781,103 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
         return motionEventList;
     }
 
+    public VideoStreamingParameters getLastCachedStreamingParameters() {
+        return parameters;
+    }
+
+    private List<VideoStreamingCapability> getSupportedCapabilities(
+            List<VideoStreamingRange> ranges,
+            List<VideoStreamingCapability> originalAdditionalCapabilities
+    ){
+
+        List<VideoStreamingCapability> validCapabilities = new ArrayList<>();
+
+        VideoStreamingCapability preferredCapability = new VideoStreamingCapability();
+        preferredCapability.setDiagonalScreenSize(parameters.getPreferredDiagonal());
+        preferredCapability.setPreferredResolution(new ImageResolution(
+                parameters.getResolution().getResolutionWidth(),
+                parameters.getResolution().getResolutionHeight())
+        );
+
+        // get the first one - the Desired resolution to guarantee streaming will start
+        validCapabilities.add(preferredCapability);
+        for (VideoStreamingRange range: ranges) {
+
+            Integer constraintHeightMax = range.getMaxResolution().getResolutionHeight();
+            Integer constraintHeightMin = range.getMinResolution().getResolutionHeight();
+            if (originalAdditionalCapabilities != null && !originalAdditionalCapabilities.isEmpty()) {
+
+                for (VideoStreamingCapability capability : originalAdditionalCapabilities) {
+                    double diagonal;
+                    if (capability.getPreferredResolution() == null
+                            || capability.getPreferredResolution().getResolutionHeight() == null
+                            || capability.getPreferredResolution().getResolutionWidth() == null) {
+                        continue;
+                    }
+                    if (capability.getDiagonalScreenSize() == null) {
+                        diagonal = parameters.getPreferredDiagonal();
+                    } else {
+                        diagonal = capability.getDiagonalScreenSize();
+                    }
+
+                    if (range.getMinScreenDiagonal() > diagonal) {
+                        continue;
+                    }
+
+                    if (!isAspectRatioInRange(range.getMinAspectRatio(), range.getMaxAspectRatio(), capability.getPreferredResolution())) {
+                        if (constraintHeightMax == null && constraintHeightMin == null) {
+                            continue;
+                        }
+                    }
+
+                    if (!isImageResolutionInRange(range.getMinResolution(), range.getMaxResolution(), capability.getPreferredResolution())) {
+                        continue;
+                    }
+
+                    validCapabilities.add(capability);
```

Comment:
Currently if a capability matches both ranges then it will be added twice to validCapabilities, We should see if the capability is already in validCapabilities before adding it.
```
if (!validCapabilities.contains(capability)) {
    validCapabilities.add(capability);
}
```